test/unit/actions/bulk_action_test.rb

```diff
@@ -0,0 +1,46 @@
+require 'foreman_tasks_test_helper'
+
+module ForemanTasks
+  class BulkActionTest < ActiveSupport::TestCase
+    before do
+      User.current = User.where(:login => 'apiadmin').first
+    end
+
+    Target = Struct.new(:id)
+
+    class ParentAction < Actions::BulkAction; end
+
+    class ChildAction < Actions::EntryAction
+      def plan(target)
+        target.id
+      end
+    end
+
+    describe Actions::BulkAction do
+      let(:targets) { (1..5).map { |i| Target.new i } }
+      let(:task) do
+        triggered = ForemanTasks.trigger(ParentAction, ChildAction, targets)
+        triggered.finished.wait(2)
```

Comment:
I think we should raise an error here when exceeding the timeout (the wait returns false in that case): When I tried running the test without dynflow changes, the test was still passing, but the task actually never finished, which I noticed by longer time until the tests actually finished (as there is the termination timeout waiting for events to finish).

Reply:
Good point, fixed